pkg/subctl/cmd/join.go

```diff
@@ -419,6 +445,35 @@ func isOpenShiftCVOEnabled(config *rest.Config) (bool, error) {
 	return scale.Spec.Replicas > 0, nil
 }
 
+func cidrIsPreConfigured(clusterID string, globalNetworks map[string]*globalnet.GlobalNetwork) bool {
+	// GlobalCIDR is not pre-configured
+	if globalNetworks[clusterID] == nil || globalNetworks[clusterID].GlobalCIDRs == nil || len(globalNetworks[clusterID].GlobalCIDRs) <= 0 {
+		return false
+	}
+	// GlobalCIDR is pre-configured
+	return true
+}
+
+func validateGlobalnetConfiguration(subctlData *datafile.SubctlData) {
+	if subctlData.GlobalnetCidrRange != "" && globalnetClusterSize != 0 && globalnetClusterSize != subctlData.GlobalnetClusterSize {
+		clusterSize, err := globalnet.GetValidClusterSize(subctlData.GlobalnetCidrRange, globalnetClusterSize)
+		if err != nil || clusterSize == 0 {
+			exitOnError("Invalid globalnet-cluster-size", err)
+		}
+		subctlData.GlobalnetClusterSize = clusterSize
+	}
+
+	if globalCIDR != "" && globalnetClusterSize != 0 {
+		err := errors.New("Both Globalnet CIDR and Globalnet Cluster size can't be specified. Specify either one")
+		exitOnError("Error:", err)
+	}
+
+	if globalCIDR != "" {
+		_, _, err := net.ParseCIDR(globalCIDR)
+		exitOnError("Specified CIDR is invalid", err)
```

Comment:
Use parameter name globalnet-cidr in error message. Something like "globalnet-cidr is invalid"